core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/DataAccessImpl.java

```diff
@@ -632,86 +655,165 @@ protected void initPreparedStatements() {
                 .concatWith(rxSession.executeAndFetch(findAllMetricsInDataCompressed.bind()));
     }
 
-    @Override
-    public Observable<Integer> insertGaugeData(Metric<Double> gauge) {
-        return insertGaugeData(gauge, -1);
+    /*
+     * Applies micro-batching capabilities by taking advantage of token ranges in the Cassandra
+     */
+    private Observable.Transformer<BoundStatement, Integer> applyMicroBatching() {
```

Comment:
Right now our insert patterns are such that we do one data point per metric.  In the event that we start doing multiple data points per metric per request, will this method group the corresponding inserts into the same batch statement?